packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -41,14 +49,31 @@ async function* loadAllSequencedMessages(
                 console.log(`reading messages${file}.json`);
                 const fileContent = fs.readFileSync(`${dir}/messages${file}.json`, { encoding: "utf-8" });
                 const messages: ISequencedDocumentMessage[] = JSON.parse(fileContent);
-                assert(messages[0].sequenceNumber === lastSeq + 1,
-                    0x1b9 /* "Unexpected value for sequence number of first message in file" */);
-                yield messages;
+                // check if there is mismatch
+                seqNumMismatch = messages[0].sequenceNumber !== lastSeq + 1;
+                assert(!seqNumMismatch, 0x1b9 /* "Unexpected value for sequence number of first message in file" */);
                 lastSeq = messages[messages.length - 1].sequenceNumber;
+                currSeq = lastSeq + 1;
+                readFromMessages = true;
+                yield messages;
             } catch (e) {
-                console.error(`Error reading / parsing messages from ${file}`);
-                console.error(e);
-                return;
+                if (seqNumMismatch) {
+                    if (overWrite) {
+                        // with overWrite option on, we will delete exisintg message.json
+                        fs.unlinkSync(`${dir}/messages${file}.json`);
```

Comment:
So when this error occurs and overwrite is true, then we need to delete all messages.json in loop rather than just 1 file otherwise we will keep having this error.